portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,26 +27,44 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> messages;
+  private ArrayList<String> names;
+  private ArrayList<String> emails; 
+  private ArrayList<String> comments;
 
+  private ArrayList<String> messages;
+ 
   @Override
   public void init() {
+    names = new ArrayList<>();
+    emails = new ArrayList<>();
+    comments = new ArrayList<>();
+
     messages = new ArrayList<>();
     messages.add("Howdy");
     messages.add("Hey, how are you?");
     messages.add("Good morning, friend!");
   }
-
+  
+  //Gets information from the server to display to the user. Picks a random message from the 
+  //messages arraylist and sends the response instance to the screen
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(messages);
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+    String message = messages.get((int) (Math.random() * messages.size()));
+    response.setContentType("text/html;");
+    response.getWriter().println(message);
   }
+  
+  //Posts information to the server to save as inputs from the data form
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name-input");
+    String email = request.getParameter("email-input");
+    String comment = request.getParameter("comment-input");
 
-  private String convertToJson(ArrayList<String> messages){
-      Gson gson = new Gson();
-      String json = gson.toJson(messages);
-      return json;
+    response.sendRedirect("http://lweiskopf-step-2020.appspot.com/contact.html");
+    
+    names.add(name);
+    emails.add(email);
+    comments.add(comment);
```

Comment:
Should you do this before the redirect?

Reply:
I can change the order around, though I believe that I delete all of this in the next pull request.